homeassistant/components/landisgyr_heat_meter/config_flow.py

```diff
@@ -0,0 +1,133 @@
+"""Config flow for Landis+Gyr Heat Meter integration."""
+from __future__ import annotations
+
+import logging
+import os
+
+import async_timeout
+import serial
+import serial.tools.list_ports
+from ultraheat_api import HeatMeterService, UltraheatReader
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MANUAL_PATH = "Enter Manually"
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ultraheat Heat Meter."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Step when setting up serial configuration."""
+        errors = {}
+
+        if user_input is not None:
+            user_selection = user_input[CONF_DEVICE]
+            if user_selection == CONF_MANUAL_PATH:
+                return await self.async_step_setup_serial_manual_path()
+
+            dev_path = await self.hass.async_add_executor_job(
+                get_serial_by_id, user_selection
+            )
+
+            try:
+                model, device_number = await self.validate_ultraheat(dev_path)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+
+            if not errors:
+                await self.async_set_unique_id(device_number)
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=model,
+                    data=user_input | {"model": model, "device_number": device_number},
+                )
```

Comment:
Move this to a function to avoid duplicating code here and in `async_step_setup_serial_manual_path`

Reply:
Done